src/Riok.Mapperly/Descriptors/MappingBodyBuilders/NewValueTupleMappingBodyBuilder.cs

```diff
@@ -0,0 +1,225 @@
+using System.Diagnostics.CodeAnalysis;
+using Microsoft.CodeAnalysis;
+using Riok.Mapperly.Abstractions;
+using Riok.Mapperly.Descriptors.MappingBodyBuilders.BuilderContext;
+using Riok.Mapperly.Descriptors.Mappings;
+using Riok.Mapperly.Descriptors.Mappings.MemberMappings;
+using Riok.Mapperly.Diagnostics;
+using Riok.Mapperly.Helpers;
+using Riok.Mapperly.Symbols;
+
+namespace Riok.Mapperly.Descriptors.MappingBodyBuilders;
+
+public static class NewValueTupleMappingBodyBuilder
+{
+    public static void BuildMappingBody(MappingBuilderContext ctx, NewValueTupleConstructorMapping expressionMapping)
+    {
+        var mappingCtx = new NewValueTupleConstructorBuilderContext<NewValueTupleConstructorMapping>(ctx, expressionMapping);
+        BuildTupleConstructorMapping(mappingCtx);
+        mappingCtx.AddDiagnostics();
+    }
+
+    public static void BuildMappingBody(MappingBuilderContext ctx, NewValueTupleExpressionMapping expressionMapping)
+    {
+        var mappingCtx = new NewValueTupleExpressionBuilderContext<NewValueTupleExpressionMapping>(ctx, expressionMapping);
+        BuildTupleConstructorMapping(mappingCtx);
+        ObjectMemberMappingBodyBuilder.BuildMappingBody(mappingCtx);
+        mappingCtx.AddDiagnostics();
+    }
+
+    private static void BuildTupleConstructorMapping(INewValueTupleBuilderContext<INewValueTupleMapping> ctx)
+    {
+        if (ctx.Mapping.TargetType is not INamedTypeSymbol namedTargetType)
+        {
+            ctx.BuilderContext.ReportDiagnostic(DiagnosticDescriptors.NoConstructorFound, ctx.BuilderContext.Target);
+            return;
+        }
+
+        if (!TryBuildTupleConstructorMapping(ctx, namedTargetType, out var constructorParameterMappings, out var mappedTargetMemberNames))
+        {
+            ctx.BuilderContext.ReportDiagnostic(DiagnosticDescriptors.NoConstructorFound, ctx.BuilderContext.Target);
+            return;
+        }
+
+        var removableMappedTargetMemberNames = mappedTargetMemberNames.Where(x =>
+        {
+            ctx.MemberConfigsByRootTargetName.TryGetValue(x, out var value);
+            return !value?.Any() ?? true;
```

Comment:
Couldn't `ContainsKey` be used here instead?

Reply:
Needs to check if the key doesn't exist or that there aren't any values in the list.

Reply:
Why do you need to check whether the list is empty? There should never be an empty list to a key. If the list gets empty it should be removed from the dictionary.